### ISSUE TITLE
Development of global, simple to use, ‘convenience’ APIs aimed at non-specialist developers

### ISSUE THREAD

**Issue by ogcscotts** [ogcscotts/TC-Meeting-topics#37]

**Comment by prushforth** [comment]
APIs are one thing, but enabling sharing using simple URLs is another. The Web is built on, and depends on the latter. So, this is the way the geospatial community should look at maps and spatial information sharing too.

We have been working on this way of sharing spatial info in developing MapML in the Maps for HTML community group, and through Testbeds 13 and 14. Here's a [demo video](https://youtu.be/qc4lLNpnA5Y) from Testbed 13 showing what we're working on. (4min)

Long story short:

- Mashups should be as easy as sharing a link
- share and develop standards with the Web
- everybody wins

APIs are for professionals.  URLs are for everybody else.

**Comment by jyutzler** [comment]
I'm not sure what MapML has to do with any of this, but I agree with the rest of your post.

**Comment by prushforth** [comment]
@jyutzler thanks for your reply

MapML is (so far) how I / we see that the geospatial information community might work with the Web community on shared standards.  I would love to hear how others propose to get maps and spatial information into the mainstream, but so far I haven't heard it, with the exception of initiatives that try to be more 'webby' in their approach.  However, I don't see server APIs, or linked data, or OGC-specific document formats (GML, KML, GeoPackage) as being more 'webby', nor as solution to the problem of getting maps and spatial information into the mainstream.   

The Web platform is mainstream, and browsers are the brokers.  Web (platform) architecture has a style, and browsers respect that style, and the undifferentiated URL is its touchstone.  So we collectively need to figure out how to standardize to that style: we need to figure out how to represent a "map" at a URL, the composition of which we don't specify.  We need to craft our standards and approach so that maps and browsers can work together. 

An ever-increasing number of server API definitions means an ever-increasing complexity for Web developers to deal with, GeoJSON notwithstanding.  MapML is trying to define basic map semantics so that the _browser_ can present geospatial APIs, not the server. If the browser is presenting the API, that is a global standard.

But to get there, we've got to work together, so maybe you could tell me how you see non-specialist developers leveraging maps and spatial information, and we can work towards what a more complete understanding of each other's viewpoint, because you say yourself that we share some common ground.

**Comment by jyutzler** [comment]
I'm having a hard time reconciling MapML as anything but another encoding for OWS Context (see #51). If OWS Context's conceptual model needs to be revised, I haven't heard any specifics. I would rather produce another encoding for an existing OGC standard than to produce something completely distinct and new. We're already fragmented enough as an industry.

**Comment by prushforth** [comment]
The point of MapML is a hypertext (which has its own conceptual model) media type that can be adopted by browsers for Web maps and spatial data, not yet another encoding of OGC standards. This is a completely different scope than that addressed by OWS Context. The HTML Web is a global standard of historically unparalleled scale so having a standard that works with the HTML Web would be unifying, not fragmenting. That's the whole point.
